src/Util/MIMETypeUtil.m

```diff
@@ -452,4 +453,1087 @@ + (NSString *)fileExtensionForMIMEType:(NSString *)mimeType
     return result;
 }
 
++ (NSDictionary *)genericMIMETypesToExtensionTypes
+{
+    static NSDictionary *result = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        result = @{
```

Comment:
Per offline conversation, this is a mishmash from many sources, manually collated.